### ISSUE TITLE
Storing temporary vowel files and simultaneous jobs

### ISSUE THREAD

**Issue by sravanareddy** [sravanareddy/FAVE#3]
FAVE-Extract currently clips each vowel portion and stores it as a temporary file with the name something-vowelname.wav. In rare cases when multiple jobs are run simultaneously and the same vowel type is clipped at the same time, this causes a conflict. A quick fix is to ensure the temporary file name is unique, but a better solution is to see if we can do this without writing temporary files at all, since it slows down the process. (FAVE-Extract is one of the performance bottlenecks of the system.)

**Comment by scjs** [comment]
Hi, I saw you were working on this! One solution that I've used to collect measurements from Praat without writing temporary sound files is to have the external script (in Python etc) create a CSV file with metadata (timestamps, unique identifiers, source file path, other measurement settings) for all of the vowel tokens that need to be measured. Then Praat can read in the CSV file, open the source wav file(s) as LongSound objects, and use the timestamps in the CSV to take the measurements in one pass.

Praat can still write out files to pass the final measurements back (like FAVE does), or I'm sure there are more efficient ways. I found that it cut running time by about half, but it also avoided a lot of freeze-ups, because Praat is only called once per run of the external script, rather than 5+ times per vowel token.

**Comment by sravanareddy** [comment]
Thanks so much for the tips! Yes, that's along the lines of what I had in mind. If we implement this into the extractFormants code directly, would that be useful for the core FAVE development as well?

**Comment by scjs** [comment]
@JoFrhwld maintains FAVE, so you could ask his thoughts? But I bet it would be appreciated, and as long as you guys put this change in its own branch, it should be pretty easy to add in.

**Comment by JoFrhwld** [comment]
Yeah, that'd be great! Ideally, you'd bundle together just the commits that make that change into on branch, and then send a pull request to the main dev branch. It'd probably be worth a +0.1 version increment.
